packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -1738,10 +1749,16 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     const headerData: HeaderData = {
       gasLimit: this.getBlockGasLimit(),
       coinbase: this.getCoinbaseAddress(),
-      nonce: "0x0000000000000042",
+      nonce: this.isPostMergeHardfork()
+        ? "0x0000000000000000"
+        : "0x0000000000000042",
       timestamp: blockTimestamp,
     };
 
+    if (this.isPostMergeHardfork()) {
+      headerData.mixHash = this._getNextMixHash();
```

Comment:
I think we should always call `_getNextMixHash` and place this HF check inside it.

The logic is wrong though, mixHash is 0 post merge according to https://eips.ethereum.org/EIPS/eip-3675

Reply:
Oh this other Paris overrides that I guess https://eips.ethereum.org/EIPS/eip-4399